### PR TITLE
factor out update_shrink_stats

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Prepare to share code.

#### Summary of Changes
Factor out metrics reporting for shrink operations.

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -3703,12 +3703,11 @@ impl AccountsDb {
 
         let mut rewrite_elapsed = Measure::start("rewrite_elapsed");
         let mut dead_storages = vec![];
-        let mut find_alive_elapsed = 0;
-        let mut create_and_insert_store_elapsed = 0;
-        let mut write_storage_elapsed = 0;
+        let mut create_and_insert_store_elapsed_us = 0;
+        let mut write_storage_elapsed = Measure::start("empty_by_default");
```
Comment (apfitzge) [inline runtime/src/accounts_db.rs:3707]: the different timing patterns here seem like an odd choice:
1. `create_and_insert_store_elapsed_us`: we use an int to accumulate time
2. `write_storage_elapsed`: initialize a dummy measure, overwrite it and stop if necessary, **relies on not calling stop to mean 0**.
3. `find_alive_elapsed`: measures a chunk of code.

Would be preferrable to have them all be consistent.

Comment (apfitzge) [reply runtime/src/accounts_db.rs:3707]: Alternatively, just make `write_storage_elapsed` an option - that seems cleaner.

**Review by apfitzge — CHANGES_REQUESTED** [review]

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs]: I've never seen us use option and for an option I can't unwrap a default. int _us seems cleaner.

**Review by apfitzge — APPROVED** [review]
Looks good

**Pull request merged**